docker-compose.yml

```diff
@@ -0,0 +1,239 @@
+      #### Redis settings ####
+      - REDIS_HOST=redis
+      - REDIS_PORT=6379
+    networks:
+      - osctrl-backend
+    depends_on:
+      - postgres
+    healthcheck:
+      test: ["CMD", "curl", "-f", "http://localhost:9002"]
+      interval: 1m30s
+      timeout: 10s
+      retries: 3
+      start_period: 40s
+
+  ######################################### PostgreSQL #########################################
+  postgres:
+    container_name: 'osctrl-postgres'
+    restart: unless-stopped
+    image: postgres:${POSTGRES_VERSION}
+    environment:
+      - POSTGRES_DB=${POSTGRES_DB_NAME}
+      - POSTGRES_USER=${POSTGRES_DB_USERNAME}
+      - POSTGRES_PASSWORD=${POSTGRES_DB_PASSWORD}
+    networks:
+      - osctrl-backend
+    volumes:
+      - postgres-db:/var/lib/postgresql/data
+    healthcheck:
+      test: ["pg_isready", "-U", "$${POSTGRES_USER}", "-d", "$${POSTGRES_DB}"]
+      interval: 10s
+      timeout: 5s
+      retries: 5
+
+  ######################################### Redis #########################################
+  redis:
+    container_name: 'osctrl-redis'
+    restart: unless-stopped
+    image: redis:${REDIS_VERSION}
+    networks:
+      - osctrl-backend
+    volumes:
+      - redis-data:/data
+
+  ######################################### osctrl-cli #########################################
+  # osctrl-cli is the component that creates the osctrl env and admin user
+  ##############################################################################################
+  osctrl-cli:
+    container_name: 'osctrl-cli'
+    restart: unless-stopped
+    image: jmpsec/osctrl-cli:v${OSCTRL_VERSION}
+    tty: true # entrypoint executes `/bin/sh`
+    environment:
+      #### osctrl env ####
+      - ENV_NAME=dev
+      - CERT_FILE=/opt/osctrl/config/osctrl.crt
+      - HOST=osctrl-nginx
+      #### osctrl admin ####
+      - OSCTRL_USER=${OSCTRL_USER}
+      - OSCTRL_PASS=${OSCTRL_PASS}
+      #### Database settings ####
+      - DB_CONFIG=true
+      - DB_JSON=/opt/osctrl/config/db.json
+      - DB_HOST=postgres
+      - DB_PORT=5432
+      - DB_NAME=${POSTGRES_DB_NAME}
+      - DB_USER=${POSTGRES_DB_USERNAME}
+      - DB_PASS=${POSTGRES_DB_PASSWORD}
+    networks:
+      - osctrl-backend
+    depends_on:
+      - postgres
+    healthcheck:
+      test: ["CMD", "/opt/osctrl/bin/osctrl-cli", "-D", "$${DB_JSON}", "check"]
+      interval: 1m30s
+      timeout: 10s
+      retries: 3
+      start_period: 60s
+
+  ######################################### osquery #########################################
+  ubuntu-osquery:
+    container_name: 'osctrl-osquery-ubuntu'
+    restart: unless-stopped
+    build:
+      context: .
+      dockerfile: deploy/docker/Dockerfile-osquery
+      args:
+        OSQUERY_VERSION: ${OSQUERY_VERSION}
+        OSCTRL_VERSION: ${OSCTRL_VERSION}
+    environment:
+      #### osquery enroll ####
+      - ENV_NAME=dev
+      - HOST=osctrl-nginx
+      #### Database settings ####
+      - DB_CONFIG=true
```

Comment:
If JSON is used, no need for env variables

Reply:
See comment in Slack